packages/flutter/lib/src/material/stepper.dart

```diff
@@ -398,11 +422,19 @@ class _StepperState extends State<Stepper> with TickerProviderStateMixin {
   }
 
   Color _circleColor(int index) {
+    final bool isActive = widget.steps[index].isActive;
     final ColorScheme colorScheme = Theme.of(context).colorScheme;
-    if (!_isDark()) {
-      return widget.steps[index].isActive ? colorScheme.primary : colorScheme.onSurface.withOpacity(0.38);
+    final Set<MaterialState> states = <MaterialState>{
+      if (isActive) MaterialState.selected else MaterialState.disabled,
+    };
+    if(widget.connectorColor?.resolve(states) != null) {
+      return widget.connectorColor!.resolve(states);
     } else {
-      return widget.steps[index].isActive ? colorScheme.secondary : colorScheme.background;
+      if (!_isDark()) {
+        return widget.steps[index].isActive ? colorScheme.primary : colorScheme.onSurface.withOpacity(0.38);
+      } else {
+        return widget.steps[index].isActive ? colorScheme.secondary : colorScheme.background;
+      }
     }
```

Comment:
final Color? color = widget.connectorColor?.resolve(states);
if (color != null) {
  return color;
}
final bool isActive = widget.steps[index].isActive;
if (!_isDark()) {
  return  isActive ? colorScheme.primary : colorScheme.onSurface.withOpacity(0.38);
}
return sActive ? colorScheme.secondary : colorScheme.background;

Reply:
it's clear now, though isActive has to be at the top because it's used in states variable.